torch/utils/data/dataset.pyi

```diff
@@ -30,7 +30,25 @@ class Dataset(Generic[T_co]):
     @classmethod
     def register_datapipe_as_function(cls, function_name: Any, cls_to_register: Any, enable_df_api_tracing: bool = ...): ...
 
-MapDataPipe = Dataset
+class MapDataPipe(Generic[T_co]):
```

Comment:
@ejguan FYI, I just made another change here. Since `IterableDataset` inherits from `Dataset`, if the methods related to `MapDataPipe` are callable by `Dataset`, then they will also be callable by `IterDataPipe` due to inheritance and that is not what we want.

For the purpose of autocompletion, I am only going to show the hints for the methods if the variable type is `MapDataPipe` but not for `Dataset`. We might have to look into changing how the inheritance works to ensure that an `IterDataPipe` won't accidentally call a method that is meant for `MapDataPipe`.